eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -191,11 +191,6 @@ public function buildDomainFields(array $spiFields, $contentType, array $languag
      */
     public function buildVersionInfoDomainObject(SPIVersionInfo $spiVersionInfo)
     {
-        $languageCodes = array();
-        foreach ($spiVersionInfo->languageIds as $languageId) {
-            $languageCodes[] = $this->contentLanguageHandler->load($languageId)->languageCode;
```

Comment:
This is what is essentially optimized away with this change, a O^N lookup of SPI languages _(they are cached by SPI Persistence Cache, so not a big win, but still)_